concurrency/buffer.go

```diff
@@ -0,0 +1,25 @@
+package concurrency
+
+import (
+	"bytes"
+	"sync"
+)
+
+type SyncBuffer struct {
```

Comment:
Can we add comment to this type?

Reply:
Yeah, I'm shocked that didn't have linter issues.